File: exercise/kotlin/day19/src/main/kotlin/blog/Blog.kt
package blog

import java.time.LocalDate

class Article(
    private val name: String,
    private val content: String,
    val comments: List<Comment> = mutableListOf()
) {
    private fun addComment(
        text: String,
        author: String,
        creationDate: LocalDate
    ): Article {
        val comment = Comment(text, author, creationDate)
        if (comments.contains(comment)) throw CommentAlreadyExistException()
        else return Article(name, content, comments + comment)
    }

    fun addComment(text: String, author: String): Article = addComment(text, author, LocalDate.now())
}


data class Comment(val text: String, val author: String, val creationDate: LocalDate)
class CommentAlreadyExistException : Exception()

File: exercise/c#/Day19/Day19/Blog.cs
using System.Collections.Immutable;

namespace Day19
{
    public class Article
    {
        private readonly string _name;
        private readonly string _content;
        public ImmutableArray<Comment> Comments { get; }

        private Article(string name, string content, IEnumerable<Comment> comments)
        {
            _name = name;
            _content = content;
            Comments = comments.ToImmutableArray();
        }

        public Article(string name, string content)
            : this(name, content, Array.Empty<Comment>())
        {
        }

        private Article AddComment(
            string text,
            string author,
            DateOnly creationDate)
        {
            var comment = new Comment(text, author, creationDate);

            return Comments.Contains(comment)
                ? throw new CommentAlreadyExistException()
                : new Article(_name, _content, Comments.Append(comment));
        }

        public Article AddComment(string text, string author)
            => AddComment(text, author, DateOnly.FromDateTime(DateTime.Now));
    }

    public record Comment(string Text, string Author, DateOnly CreationDate);

    public class CommentAlreadyExistException : ArgumentException;
}